lib/spack/spack/reporters/cdash.py

```diff
@@ -248,16 +295,22 @@ def upload(self, filename):
             # CDash needs expects this file to be uploaded via PUT.
             request.get_method = lambda: 'PUT'
             response = opener.open(request)
-            if not self.buildId:
-                match = buildid_regexp.search(response.read())
+            if self.current_package_name not in self.buildIds:
+                match = self.buildid_regexp.search(response.read())
                 if match:
-                    self.buildId = match.group(1)
+                    buildid = match.group(1)
+                    self.buildIds[self.current_package_name] = buildid
 
     def print_cdash_link(self):
-        if self.buildId:
-            # Construct and display a helpful link if CDash responded with
-            # a buildId.
-            build_url = self.cdash_upload_url
-            build_url = build_url[0:build_url.find("submit.php")]
-            build_url += "buildSummary.php?buildid={0}".format(self.buildId)
-            print("View your build results here:\n  {0}\n".format(build_url))
+        if self.buildIds:
+            print("View your build results here:")
+            for package_name, buildid in iteritems(self.buildIds):
+                # Construct and display a helpful link if CDash responded with
+                # a buildId.
+                build_url = self.cdash_upload_url
+                build_url = build_url[0:build_url.find("submit.php")]
+                build_url += "buildSummary.php?buildid={0}".format(buildid)
+                if self.multiple_packages:
+                    print("{0}: {1}".format(package_name, build_url))
+                else:
+                    print("{0}".format(build_url))
```

Comment:
Just curious if it makes sense to always report the `pkgname: url`?

Reply:
👍